File: 0206-reverse-linked-list/0206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        if(head==null)
            return null;
        ListNode cur=head,right=head,left=null;
        while(cur!=null)
        {
            right=cur.next;
            cur.next=left;
            left=cur;
            cur=right;
        }
        return left;
    }
}

File: 0206-reverse-linked-list/0206-reverse-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseList(struct ListNode* head){
    struct ListNode* cur = head;
    struct ListNode* fow = head;
    struct ListNode* prev = NULL;
    while(fow != NULL)
    {
        fow = fow->next;
        cur->next = prev;
        prev = cur;
        cur = fow;
    }
    return prev;
}